### PR TITLE
config: update apex checks to support Checkstyle 8.29

### PR DESCRIPTION
**Pull request by nmancus1**
bump checkstyle version to support maven-checkstyle-plugin

### CODE REVIEW

File: codestyle-config/src/main/resources/apex_checks.xml [hunk 1]
```diff
@@ -19,9 +19,9 @@
     under the License.
 
 -->
-<!DOCTYPE module PUBLIC
-  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
-  "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+    <!DOCTYPE module PUBLIC
+        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
```
Comment (rnveach) [inline codestyle-config/src/main/resources/apex_checks.xml]: Please fix unnecessary change issues. I see extra indentation being added or removed in this file.

File: codestyle-config/src/main/resources/apex_checks.xml [hunk 2]
```diff
@@ -117,7 +117,7 @@
     <module name="ModifierOrder"/>
     <module name="EmptyLineSeparator">
       <property name="tokens"
-                value="IMPORT, CLASS_DEF, ENUM_DEF, INTERFACE_DEF, CTOR_DEF, METHOD_DEF, STATIC_INIT, INSTANCE_INIT"/>
+          value="IMPORT, CLASS_DEF, ENUM_DEF, INTERFACE_DEF, CTOR_DEF, METHOD_DEF, STATIC_INIT, INSTANCE_INIT"/>
```
Comment (rnveach) [inline codestyle-config/src/main/resources/apex_checks.xml]: Same.

File: codestyle-config/src/main/resources/apex_checks.xml (hunk 1 above)
Comment (nmancus1) [reply codestyle-config/src/main/resources/apex_checks.xml]: Fixed.

File: codestyle-config/src/main/resources/apex_checks.xml (hunk 2 above)
Comment (nmancus1) [reply codestyle-config/src/main/resources/apex_checks.xml]: Fixed.

**Comment by nmancus1** [comment]
> Please confirm you ran regression commands locally on this repo and it passed without any issues.

Still have Checkstyle violations, LineLength and other errors have been resolved.

**Comment by rnveach** [comment]
> Still have Checkstyle violations, LineLength and other errors have been resolved

I am confused. Running the regression commands on this PR shouldn't produce any violations as this repo is used as no error regression. It means checkstyle completed and found no violations.

If there are still violations, then they would have to be fixed for this to still be considered a no error regression.

File: codestyle-config/src/main/resources/apex_checks.xml [hunk 3]
```diff
@@ -117,7 +117,7 @@
     <module name="ModifierOrder"/>
     <module name="EmptyLineSeparator">
       <property name="tokens"
-                value="IMPORT, CLASS_DEF, ENUM_DEF, INTERFACE_DEF, CTOR_DEF, METHOD_DEF, STATIC_INIT, INSTANCE_INIT"/>
```
Comment (rnveach) [inline codestyle-config/src/main/resources/apex_checks.xml]: Still more indentation issues. Please fix all.

**Comment by nmancus1** [comment]
@rnveach Here is the output: https://pastebin.com/wtuCirKn

**Comment by nmancus1** [comment]
Here is Checkstyle output: https://pastebin.com/wtuCirKn

**Comment by rnveach** [comment]
> Here is Checkstyle output: https://pastebin.com/wtuCirKn

Please resolve them all. If you think it will take time to resolve them, let me know and we can just merge this PR in the meantime. 

Either way, please fix indentation issues in this PR so they don't show as differences unless they really are a difference.

**Comment by nmancus1** [comment]
> Please resolve them all. If you think it will take time to resolve them, let me know and we can just merge this PR in the meantime.

I can get in done within two days, for sure.  I have midterms this week and next, so time is a little short for me. 

> Either way, please fix indentation issues in this PR so they don't show as differences unless they really are a difference.

I apologize, I used Intellij to reformat, it must've shifted everything around. I'll fix it again.

**Comment by romani** [comment]
> ImportOrder: Extra separation in import group 

sounds like "easy" fix might be. Please try to fix and send PR.

**Comment by nmancus1** [comment]
> > ImportOrder: Extra separation in import group
> 
> sounds like "easy" fix might be. Please try to fix and send PR. It is better to do fixes manually(in simple editor) to not let IDEA do more on your behalf.

That makes a lot of sense.  IDEA is too complicated sometimes, I think.

**Comment by nmancus1** [comment]
@romani I have fixed all of the checkstyle violations, should I make a new PR for them?

**Comment by nmancus1** [comment]
After I corrected the first 36 Checkstyle violations, 50 popped up, then 736. Is it worth fixing at this point?

**Comment by romani** [comment]
>Is it worth fixing at this point?

No, it doesn't make sense.
I am to drop this project from CI at all.
